[2024-07-25T12:24:29.181+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-25T12:24:29.281+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_etl_pipeline.ingest_csv scheduled__2024-07-23T00:00:00+00:00 [queued]>
[2024-07-25T12:24:29.310+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_etl_pipeline.ingest_csv scheduled__2024-07-23T00:00:00+00:00 [queued]>
[2024-07-25T12:24:29.311+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2024-07-25T12:24:29.373+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): ingest_csv> on 2024-07-23 00:00:00+00:00
[2024-07-25T12:24:29.388+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:62: DeprecationWarning: This process (pid=66) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-07-25T12:24:29.391+0000] {standard_task_runner.py:64} INFO - Started process 790 to run task
[2024-07-25T12:24:29.397+0000] {clientserver.py:543} INFO - Closing down clientserver connection
[2024-07-25T12:24:29.404+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spark_etl_pipeline', 'ingest_csv', 'scheduled__2024-07-23T00:00:00+00:00', '--job-id', '125', '--raw', '--subdir', 'DAGS_FOLDER/spark_demo.py', '--cfg-path', '/tmp/tmpazdm115u']
[2024-07-25T12:24:29.405+0000] {standard_task_runner.py:91} INFO - Job 125: Subtask ingest_csv
[2024-07-25T12:24:29.599+0000] {task_command.py:426} INFO - Running <TaskInstance: spark_etl_pipeline.ingest_csv scheduled__2024-07-23T00:00:00+00:00 [running]> on host 1671a0defda5
[2024-07-25T12:24:29.876+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='your_email@example.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_etl_pipeline' AIRFLOW_CTX_TASK_ID='ingest_csv' AIRFLOW_CTX_EXECUTION_DATE='2024-07-23T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-07-23T00:00:00+00:00'
[2024-07-25T12:24:29.878+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-25T12:24:53.934+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-25T12:24:53.935+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/spark_demo.py", line 18, in ingest_csv
    df_csv = df_csv.withColumn('source', 'csv')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/dataframe.py", line 4785, in withColumn
    raise PySparkTypeError(
pyspark.errors.exceptions.base.PySparkTypeError: [NOT_COLUMN] Argument `col` should be a Column, got str.
[2024-07-25T12:24:53.974+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=spark_etl_pipeline, task_id=ingest_csv, run_id=scheduled__2024-07-23T00:00:00+00:00, execution_date=20240723T000000, start_date=20240725T122429, end_date=20240725T122453
[2024-07-25T12:24:54.041+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 125 for task ingest_csv ([NOT_COLUMN] Argument `col` should be a Column, got str.; 790)
[2024-07-25T12:24:54.114+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2024-07-25T12:24:54.144+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-07-25T12:24:54.150+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
